controllers/etcdcluster_controller.go

```diff
@@ -404,13 +405,15 @@ func (r *EtcdClusterReconciler) reconcile(
 				}
 			}
 
-			// Upgrade Etcd.
+			// Upgrade Etcd or incorporate changes.
 			// Remove EtcdPeers which  have a different version than EtcdCluster; in reverse name order, one-at-a-time.
-			// The EtcdPeers will be recreated in the next reconcile, with the correct version.
-			if peer := nextOutdatedPeer(cluster, peers); peer != nil {
-				if peer.Spec.Version == cluster.Spec.Version {
-					log.Info("Waiting for EtcdPeer to report expected server version", "etcdpeer-name", peer.Name)
-					return result, nil, nil
+			// The EtcdPeers will be recreated in the next reconcile, with the correct state.
+			if peer, reason := nextOutdatedPeer(cluster, peers); peer != nil {
+				if reason == outdatedVersion {
+					if peer.Spec.Version == cluster.Spec.Version {
+						log.Info("Waiting for EtcdPeer to report expected server version", "etcdpeer-name", peer.Name)
```

Comment:
Maybe put a comment here to elaborate. Just something to explain that this branch occurs when we've updated the Pod specification but the etcd peer still shows as outdated.

Reply:
done